### PR TITLE
Add extra channel for critical events

### PR DESCRIPTION
**Pull request by annakocheshkova**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [x] Did you add unit tests?
* [X] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Adds two extra channel for Analytics critical events.

No unit tests yet.

## Related PRs or issues

[AB#61312](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/61312)
[PR#1602](https://github.com/microsoft/AppCenter-SDK-Apple/pull/1602)

### CODE REVIEW

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m [hunk 1]
```diff
@@ -813,6 +815,31 @@ - (void)testTrackEventWithPropertiesWithInvalidFlag {
...
+  OCMExpect([self.channelUnitCriticalMock enqueueItem:OCMOCK_ANY flags:MSFlagsPersistenceCritical]);
+  OCMExpect([self.channelUnitMock enqueueItem:OCMOCK_ANY flags:MSFlagsPersistenceNormal]);
+
+  // When
+  [[MSAnalytics sharedInstance] trackEvent:expectedCriticalEvent
+                       withTypedProperties:nil
+                     forTransmissionTarget:nil
+                                     flags:MSFlagsPersistenceCritical];
+  [[MSAnalytics sharedInstance] trackEvent:expectedEvent withTypedProperties:nil forTransmissionTarget:nil flags:MSFlagsPersistenceNormal];
+  // Then
```
Comment (MatkovIvan) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: Add empty line

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m [hunk 2]
```diff
@@ -813,6 +815,31 @@ - (void)testTrackEventWithPropertiesWithInvalidFlag {
...
+
+  // If
+  NSString *expectedCriticalEvent = @"Having a cup of coffee";
+  NSString *expectedEvent = @"Washing a cup after having a coffee";
+  [MSAppCenter configureWithAppSecret:kMSTestAppSecret];
+  [[MSAnalytics sharedInstance] startWithChannelGroup:self.channelGroupMock
+                                            appSecret:kMSTestAppSecret
+                              transmissionTargetToken:nil
+                                      fromApplication:YES];
+  [MSAnalytics sharedInstance].criticalChannelUnit = self.channelUnitCriticalMock;
```
Comment (MatkovIvan) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: Move it to setup

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m [hunk 3]
```diff
@@ -82,6 +83,7 @@ - (void)setUp {
   // Mock channel.
   self.channelGroupMock = OCMClassMock([MSChannelGroupDefault class]);
   self.channelUnitMock = OCMProtocolMock(@protocol(MSChannelUnitProtocol));
+  self.channelUnitCriticalMock = OCMProtocolMock(@protocol(MSChannelUnitProtocol));
   OCMStub([self.channelGroupMock alloc]).andReturn(self.channelGroupMock);
   OCMStub([self.channelGroupMock initWithInstallId:OCMOCK_ANY logUrl:OCMOCK_ANY]).andReturn(self.channelGroupMock);
   OCMStub([self.channelGroupMock addChannelUnitWithConfiguration:OCMOCK_ANY]).andReturn(self.channelUnitMock);
```
Comment (MatkovIvan) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m]: Update this to return `channelUnitCriticalMock` if when it requested

File: AppCenterAnalytics/AppCenterAnalytics/Internals/MSAnalyticsPrivate.h [hunk 4]
```diff
@@ -10,6 +10,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+// The Id for critical events.
+static NSString *const kMSCriticalChannelPostfix = @"critical";
```
Comment (jwallra) [inline AppCenterAnalytics/AppCenterAnalytics/Internals/MSAnalyticsPrivate.h]: I noticed that another variable used later uses "suffix": `kMSOneCollectorGroupIdSuffix`

We should align this with preexisting names and use `kMSCriticalChannelSuffix`

File: AppCenterAnalytics/AppCenterAnalytics/Internals/MSAnalyticsPrivate.h [hunk 5]
```diff
@@ -10,6 +10,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+// The Id for critical events.
```
Comment (Jamminroot) [inline AppCenterAnalytics/AppCenterAnalytics/Internals/MSAnalyticsPrivate.h]:
```suggestion
// The Id suffix for critical events.
```

**Pull request merged**